modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.tsx

```diff
@@ -50,442 +88,744 @@ const getVisibleFields = ({
+														itemId={id}
+														itemInlineChanges={
+															itemInlineChanges
+														}
+														rootPropertyName={
+															localizedValue?.rootPropertyName ??
+															undefined
+														}
+														value={
+															localizedValue?.value ??
+															undefined
+														}
+														valuePath={valuePath}
+													/>
+												</ClayTableCell>
+											);
+										}
+									}
+								}
+							}
+						</ClayTableRow>
+					);
+				}
+			}
+		</ClayTableBody>
 	);
 };
 
-const Body = ({
-	highlightedItemsValue,
-	inlineAddingSettings,
-	items,
-	itemsActions,
-	itemsChanges,
-	nestedItemsKey,
-	nestedItemsReferenceKey,
-	selectItems,
-	selectable,
-	selectedItemsKey,
-	selectedItemsValue,
-	selectionType,
-	visibleFields,
-}: {
-	highlightedItemsValue: any;
-	inlineAddingSettings: any;
-	items: Array<any>;
-	itemsActions: Array<IItemsActions>;
-	itemsChanges: any;
-	nestedItemsKey: string | undefined;
-	nestedItemsReferenceKey: string | undefined;
-	selectItems: Function;
-	selectable: boolean | undefined;
-	selectedItemsKey: string;
-	selectedItemsValue: any;
-	selectionType: string | undefined;
-	visibleFields: Array<any>;
-}) => {
-	const actionExists = Boolean(
-		itemsActions?.length ||
-			items?.find((item) => item.actionDropdownItems?.length)
+/**
+ * Wrapper on top of ClayCell to add column resizer capabilities. This
+ * should be removed when Clay implements this feature with accessibility.
+ */
+function HeadCellResizer({
+	children,
+	columnName,
+	...otherProps
+}: React.ComponentProps<typeof ClayTableCell> & {
+	columnName: string;
+}) {
+	const {
+		draggingAllowed,
+		draggingColumnName,
+		isFixed,
+		resizeColumn,
+		updateDraggingAllowed,
+		updateDraggingColumnName,
+	} = useContext(TableContext);
+
+	const [{modifiedFields}, viewsDispatch]: [
+		IViewsContext,
+		TViewsContextDispatch,
+	] = useContext(ViewsContext);
+
+	const cellRef = useRef<HTMLTableCellElement>(null);
+	const clientXRef = useRef({current: null});
+
+	useEffect(() => {
+		if (columnName && !isFixed && cellRef.current) {
+			const boundingClientRect = cellRef.current.getBoundingClientRect();
+
+			viewsDispatch({
+				type: VIEWS_ACTION_TYPES.UPDATE_FIELD,
+				value: {
+					name: columnName,
+					resizable: true,
+					width: boundingClientRect.width,
+				},
+			});
+		}
+	}, [columnName, isFixed, viewsDispatch]);
+
+	const handleDrag = useMemo(() => {
+
+		// eslint-disable-next-line react-compiler/react-compiler
+		return throttle((event) => {
+			if (event.clientX === clientXRef.current || !cellRef.current) {
+				return;
+			}
+
+			updateDraggingColumnName(columnName);
+
+			clientXRef.current = event.clientX;
+
+			const {x: headerCellX} = cellRef.current.getClientRects()[0];
+			const newWidth = event.clientX - headerCellX;
+
+			resizeColumn(columnName, newWidth);
+		}, 20);
+	}, [columnName, resizeColumn, updateDraggingColumnName]);
```

Comment:
Disabling eslint here is a temporary solution. Formatter is right here that this is a bad practice. From https://react.dev/reference/react/useRef#caveats:
> Do not write or read ref.current during rendering, except for [initialization.](https://react.dev/reference/react/useRef#avoiding-recreating-the-ref-contents) This makes your component’s behavior unpredictable

I think the whole solution for dragging needs rework. [In DSM](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx) we used `react-dnd` library for dragging, that avoids various React pitfalls. That could be bases of solution here, if it becomes too complicated. I think fixing this can be a separate task.

A sidenote, @bryceosterhaus is it intentional that SF locally succeeds, even with the error? 
```
/home/marko/git/lp-master/portal/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/Table.tsx
  480:19  error  Ref values (the `current` property) may not be accessed during render. (https://react.dev/reference/react/useRef)  react-compiler/react-compiler

✖ 1 problem (1 error, 0 warnings)

Done in 9.49s.

BUILD SUCCESSFUL in 35s
4 actionable tasks: 4 executed
```

Reply:
> is it intentional that SF locally succeeds, even with the error?

What do you mean here? What command passes and which one fails?

We do have some errors we ignore because they were originally here when we added the compiler rule

Reply:
In the latest PR, I added `// eslint-disable-next-line react-compiler/react-compiler` and everything is green with SF.

Without it, there is a legitimate SF error, that I quoted above. It's a valid error, I go over the root cause and what to do above. What I am noting is that local `gradle formatSource` still succeeded, despite error (see console output above). This seems odd to me, not sure if it's expected.

Reply:
> I think fixing this can be a separate task.

That is fine, can we just be sure to create the task and try and address it sooner than later? Otherwise other teams will grep for the error and copy the eslint-disable

As for the gradle sf passing even with the error... that is odd. I will pull down your changes and check it out